sample/sample8-middlewares/UserController.ts

```diff
@@ -0,0 +1,56 @@
+import {Request} from "express";
+import {Controller} from "../../src/decorator/controllers";
+import {Get, Post, Put, Patch, Delete, Use} from "../../src/decorator/methods";
+import {Req, Next} from "../../src/decorator/params";
+
+@Controller()
+export class UserController {
+
+
+    @Use()
+    middleware( @Next() next: any) {
+      console.log("calling middleware");
+      next();
+    }
+
+    @Get("/users", { jsonResponse: true })
+    getAll() {
+        return [
+            { id: 1, name: "First user!" },
+            { id: 2, name: "Second user!" }
+        ];
+    }
+
+    @Get("/users/:id")
+    getOneMdlw( @Req() request: Request, @Next() next: any) {
```

Comment:
I agree, it tied it up with express, perhaps we can specify some specific decorator in server implementations, to that they're only avalaible when a specific server framework is used.

I did a quick search for next function in koa, it seem they use it the same way... for the others I really don't know :)!

If you have suggestions?